zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/SnapStream.java

```diff
@@ -104,10 +104,20 @@ public static CheckedInputStream getInputStream(File file) throws IOException {
         InputStream is;
         switch (getStreamMode(file.getName())) {
         case GZIP:
-            is = new GZIPInputStream(fis);
+            try {
```

Comment:
Put the entire block in a try-catch:
```java
try {
        switch (getStreamMode(file.getName())) {
        case GZIP:
            is = new GZIPInputStream(fis);
            break;
        case SNAPPY:
            is = new SnappyInputStream(fis);
            break;
        case CHECKED:
        default:
            is = new BufferedInputStream(fis);
        }
        return new CheckedInputStream(is, new Adler32());
} catch (IOException e) {
    fis.close();
    throw e;
}
```

Reply:
I've made this change. Would you all also prefer one `try` around the entire `switch` in the other method, where only one of the `case`s can throw? It would make the code look more symmetrical, but it's otherwise unnecessary.